### PR TITLE
Test curations with empty name / version

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: model/src/test/kotlin/PackageCurationTest.kt [hunk 1]
```diff
@@ -409,6 +410,22 @@ class PackageCurationTest : WordSpec({
     }
 
     "isApplicable()" should {
+        "accept empty an name and / or version" {
```
Comment (mnonnenmacher) [inline model/src/test/kotlin/PackageCurationTest.kt:413]: Typo: "an empty"

File: model/src/test/kotlin/PackageCurationTest.kt [hunk 2]
```diff
@@ -409,6 +410,22 @@ class PackageCurationTest : WordSpec({
...
+                data = PackageCurationData(
+                    authors = sortedSetOf("The Android Open Source Project"),
+                    concludedLicense = "Apache-2.0".toSpdx()
+                )
+            )
+
+            assertSoftly {
+                curation.isApplicable(Identifier("Maven:com.android.tools:common:25.3.0")) shouldBe true
+                curation.isApplicable(Identifier("Maven:com.android.tools:common")) shouldBe true
+                curation.isApplicable(Identifier("Maven:com.android.tools")) shouldBe true
```
Comment (mnonnenmacher) [inline model/src/test/kotlin/PackageCurationTest.kt:425]: Add `curation.isApplicable(Identifier("Maven:com.android.tools::25.2.0")) shouldBe true` to test all cases (IIRC).

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Review by mnonnenmacher — APPROVED** [review]

File: model/src/test/kotlin/PackageCurationTest.kt [hunk 3]
```diff
@@ -409,6 +410,23 @@ class PackageCurationTest : WordSpec({
     }
 
     "isApplicable()" should {
+        "accept an empty name and / or version" {
+            val curation = PackageCuration(
+                id = Identifier("Maven:com.android.tools"),
+                data = PackageCurationData(
+                    authors = sortedSetOf("The Android Open Source Project"),
```
Comment (fviernau) [inline model/src/test/kotlin/PackageCurationTest.kt:417]: Setting authors and concluded license is not needed and should be dropped.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: model/src/test/kotlin/PackageCurationTest.kt (hunk 3 above)
Comment (sschuberth) [reply model/src/test/kotlin/PackageCurationTest.kt:417]: Actually, I added this deliberately even if not strictly needed to showcase the idea described in the [comment linked from the commit message](https://github.com/oss-review-toolkit/ort/issues/5105#issuecomment-1108228824). I'll also add a code comment.

Comment (fviernau) [reply model/src/test/kotlin/PackageCurationTest.kt:421]: The test is about the matching, not about overriding the `authors` and `concluded license` property. 
I doubt the reader of the test will be aware of the comment link you provided above.
So, from the perspective of that reader the data would be irrelevant as well I guess.

**Review by fviernau — DISMISSED** [review]

**Review by fviernau — APPROVED** [review]

**Pull request merged**